### PR TITLE
Feature latest

### PR DESCRIPTION
**Pull request by VivekHande16**
Changes done in this PR are:

Added login/signup functionality.
Added add Salons data feature.
Added Get Salons data feature.
Setup GitHub Actions CI/CD.

### CODE REVIEW

File: src/database/db_func.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        } else {
+          sequelize.sync();
+        }
+      }
+    })
+  ).catch((error) => {
+    logger.error(error);
+    return;
+  });
+
```
Comment (jesusverma) [inline src/database/db_func.js:67]: Object.keys(sequelize.models).forEach((key) => {
    if ('associate' in sequelize.models[key]) {
      sequelize.models[key].associate(sequelize.models);
    }
  });
  
  
  for asscoiation with different models

File: src/server/router/salon.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const { Router } = require('express');
+const Status = require('http-status');
+
+const salonCreate = require('../../command/SalonCreate');
+const salonGetAll = require('../../assembler/SalonGetAll');
+const salonGetFew = require('../../assembler/SalonGetFew');
```
Comment (jesusverma) [inline src/server/router/salon.js]: salonGetFew : salonGet

File: src/server/router/app.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        },
+        promClient: {
+          collectDefaultMetrics: {
+            timeout: 1000,
+          },
+        },
+      })
+    );
+
+  app.get('/healthz/liveness', function health(req, res) {
```
Comment (jesusverma) [inline src/server/router/app.js:41]: check this healthcheck

File: src/domain/repository/user.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      transaction: t,
+    });
+
+    let users = new_user.map((k) => toDomain(k));
+    return users;
+  };
+
+  const getOne = async (user, t) => {
+    const new_user = await database.models.user.findOne({
+      where: { Email: user.Email },
```
Comment (jesusverma) [inline src/domain/repository/user.js]: Email : email

File: src/domain/repository/user.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    const new_user = await database.models.user.findAll({
+      //   where: constraint,
+      transaction: t,
+    });
+
+    let users = new_user.map((k) => toDomain(k));
+    return users;
+  };
+
+  const getOne = async (user, t) => {
```
Comment (jesusverma) [inline src/domain/repository/user.js]: getOne -> login

File: src/domain/repository/user.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    const new_user = await database.models.user.findOne({
+      where: { Email: user.Email },
+      transaction: t,
+    });
+
+    const { dataValues } = new_user;
+    let bool = bcrypt.compareSync(user.Password, dataValues.Password);
+    if (!bool) return toDomain(null);
```
Comment (jesusverma) [inline src/domain/repository/user.js]: Send a proper msg

File: src/domain/repository/user.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    return toDomain(new_user);
+  };
+
+  const toDomain = ({ dataValues }) => {
+    return new User({
+      UserId: dataValues.UserId,
```
Comment (jesusverma) [inline src/domain/repository/user.js]: smallCase or camel case or underscore_case

Comment (jesusverma) [reply src/domain/repository/user.js]: ManPower -> man_power / manPower

File: src/domain/repository/salon.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+      transaction: t,
+    });
+
+    let salons = new_salon.map((k) => toDomain(k));
+    return salons;
+  };
+
+  const getFew = async (salon, t) => {
+    const new_salon = await database.models.salon.findAll({
+      where: { City: salon.City },
```
Comment (jesusverma) [inline src/domain/repository/salon.js]: searchByKey  : city
searchByValue :  Banglore

salon : data , query , payload

Comment (jesusverma) [reply src/domain/repository/salon.js]: where :{
[searchByKey] : searchByValue
 }

Comment (jesusverma) [reply src/domain/repository/salon.js]: put the data filter

File: src/domain/models/user.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+const { attributes } = require('structure');
+
+const User = attributes({
+  UserId: { type: String, required: true },
```
Comment (jesusverma) [inline src/domain/models/user.js]: length:  40 constraint

File: src/command/UserCreate.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+  const saltRounds = 10;
+
+  const salt = bcrypt.genSaltSync(saltRounds);
+  const hash = bcrypt.hashSync(payload.Password, salt);
+  payload.Password = hash;
+
+  const UID = CoreUtil.randomFTEID('UID');
+  payload.UserId = UID;
+  const user = new User(payload);
```
Comment (jesusverma) [inline src/command/UserCreate.js]: handle error

File: src/command/UserCreate.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+const bcrypt = require('bcrypt');
+
+module.exports = async (payload, context, t, repository) => {
+  const { userRepository } = repository;
+
+  const saltRounds = 10;
+
+  const salt = bcrypt.genSaltSync(saltRounds);
+  const hash = bcrypt.hashSync(payload.Password, salt);
+  payload.Password = hash;
```
Comment (jesusverma) [inline src/command/UserCreate.js]: put this logic in a func

File: src/command/UserCreate.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
+const User = require('../domain/models/user');
+const CoreUtil = require('../utils/core');
+const bcrypt = require('bcrypt');
+
+module.exports = async (payload, context, t, repository) => {
+  const { userRepository } = repository;
+
+  const saltRounds = 10;
```
Comment (jesusverma) [inline src/command/UserCreate.js]: value of saltRounds should go in env

File: package.json [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    "eslint": "^8.21.0",
+    "eslint-config-node": "^4.1.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-prettier": "^4.2.1",
+    "nodemon": "^2.0.19",
+    "prettier": "^2.7.1"
+  },
+  "engines": {
+    "node": ">=10.0.0"
```
Comment (jesusverma) [inline package.json]: 16

File: src/database/db_func.js (hunk 1 above)
Comment (VivekHande16) [reply src/database/db_func.js:67]: Added the above code.

File: src/server/router/salon.js (hunk 2 above)
Comment (VivekHande16) [reply src/server/router/salon.js]: done

File: package.json (hunk 13 above)
Comment (VivekHande16) [reply package.json]: done

File: src/command/UserCreate.js (hunk 12 above)
Comment (VivekHande16) [reply src/command/UserCreate.js]: done

File: src/domain/models/user.js (hunk 9 above)
Comment (VivekHande16) [reply src/domain/models/user.js]: done

File: src/domain/repository/salon.js (hunk 8 above)
Comment (VivekHande16) [reply src/domain/repository/salon.js]: done

File: src/domain/repository/user.js (hunk 5 above)
Comment (VivekHande16) [reply src/domain/repository/user.js]: done

File: src/domain/repository/user.js (hunk 7 above)
Comment (VivekHande16) [reply src/domain/repository/user.js]: done

File: src/server/router/app.js (hunk 3 above)
Comment (VivekHande16) [reply src/server/router/app.js:41]: done

File: src/domain/repository/user.js (hunk 4 above)
Comment (VivekHande16) [reply src/domain/repository/user.js]: done

File: src/domain/repository/user.js (hunk 6 above)
Comment (VivekHande16) [reply src/domain/repository/user.js]: done

**Pull request merged**